unixcreds.go

```diff
@@ -51,14 +52,20 @@ func UnixSocketRequireRoot() UnixCredentialsFunc {
 	return UnixSocketRequireUidGid(0, 0)
 }
 
-// UnixSocketRequireSameUser resolves the current unix user and returns a
+// UnixSocketRequireSameUser resolves the current effective unix user and returns a
 // UnixCredentialsFunc that will validate incoming unix connections against the
 // current credentials.
 //
 // This is useful when using abstract sockets that are accessible by all users.
+//
+// This function validates the *effective* UID/GID rather than the real UID/GID.
```

Comment:
Move this description to `UnixSocketRequireUidGid` and point to it here instead.